core/vulkan/loader/loader.go

```diff
@@ -69,46 +156,51 @@ func SetupTrace(ctx context.Context, env *shell.Env) (func(), string, error) {
 
 // SetupReplay sets up the environment for local replay. Returns a clean-up
 // function to be called after replay completes, or an error.
-func SetupReplay(ctx context.Context, env *shell.Env) (func(), error) {
-	lib, json, err := findLibraryAndJSON(ctx, layout.LibVirtualSwapChain)
+func SetupReplay(ctx context.Context, d bind.Device, abi *device.ABI, env *shell.Env) (func(ctx context.Context), error) {
+	var setup DeviceReplaySetup
+	if dev, ok := d.(remotessh.Device); ok {
+		setup = &remoteSetup{dev, abi}
+	} else {
+		setup = &localSetup{abi}
+	}
+	tempdir, cleanup, err := setup.MakeTempDir(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	lib, json, err := findLibraryAndJSON(ctx, setup, tempdir, layout.LibVirtualSwapChain)
 	if err != nil {
-		return func() {}, err
+		return func(ctx context.Context) {}, err
 	}
-	return setupJSON(lib, json, env)
+
+	err = setupJSON(ctx, lib, json, setup, tempdir, env)
+	return cleanup, err
 }
 
-func findLibraryAndJSON(ctx context.Context, libType layout.LibraryType) (file.Path, file.Path, error) {
-	lib, err := layout.Library(ctx, libType)
+func findLibraryAndJSON(ctx context.Context, rs DeviceReplaySetup, tempdir string, libType layout.LibraryType) (string, file.Path, error) {
```

Comment:
Although this is private, would be great to document the return values.

Reply:
Done